tests/test_csv.py

```diff
@@ -0,0 +1,590 @@
+    ]
+
+
+def test_continuable():
+    """Make sure we can continue parsing after handling a non-fatal error."""
+    text = dedent(
+        """\
+        address|type
+        5555|libary
+        1234|function
+        zzzz|function
+        4321|template
+        """
+    )
+
+    # Should throw for "libary"
+    with pytest.raises(ReccmpCsvParserError):
+        list(csv_parse(text))
+
+    # Parse the same text but catch the exception
+    values = []
+    reader = csv_parse(text)
+    while True:
+        try:
+            values.append(next(reader))
+        except StopIteration:
+            break
+        except ReccmpCsvParserError:
+            continue
+
+    # Should exclude the two failed values and return the ones we can parse.
+    assert values == [
+        (0x1234, {"type": EntityType.FUNCTION}),
+        (0x4321, {"type": EntityType.FUNCTION}),
+    ]
+
+
+def test_exception_details():
+    """Should capture the original line number (before removing blank lines)
+    and the value that led to the exception and report both."""
+    text = dedent(
+        """\
+        address|type
+
+        5555|libary
+        1234|function
+
+        zzzz|function
+        4321|template
+        """
+    )
+
+    count = 0
+    reader = csv_parse(text)
+    while True:
+        try:
+            next(reader)
+        except StopIteration:
+            break
+        except CsvInvalidAddressError as ex:
+            assert ex.illegal_value == "zzzz"
+            assert ex.line_number == 6
+            count += 1
+            continue
+        except CsvInvalidEntityTypeError as ex:
+            assert ex.illegal_value == "libary"
+            assert ex.line_number == 3
+            count += 1
+            continue
+
+    assert count == 2
+
+
+def test_docs_example_basic():
+    """Just make sure we can parse it"""
+    text = dedent(
+        """\
+        address|name|size
+        1008b400|_atol|164
+        1008b4b0|_atoi|14
+        1008b4c0|_strtok|216
+        1008b5a0|_sprintf|103
+        1008b608|__ftol|39"""
+    )
+    assert (0x1008B400, {"name": "_atol", "size": 164}) in list(csv_parse(text))
+
+
+def test_docs_example_null_field():
+    """0x10001070 has null type"""
+    text = dedent(
+        """\
+        address|type|size
+        0x10001000|function|92
+        0x10001070||25
+        0x10001090|function|10"""
+    )
+    assert (0x10001070, {"size": 25}) in list(csv_parse(text))
+
+
+def test_docs_example_quoted_field():
+    """Can parse double-quote-wrapped field that contains delimiter"""
+    text = dedent(
+        """\
+        addr,name
+        101310a0,\"set<MxAtom *,MxAtomCompare,allocator<MxAtom *> >::set<MxAtom *,MxAtomCompare,allocator<MxAtom *> >\""""
+    )
+    assert (
+        0x101310A0,
+        {
+            "name": "set<MxAtom *,MxAtomCompare,allocator<MxAtom *> >::set<MxAtom *,MxAtomCompare,allocator<MxAtom *> >"
+        },
+    ) in list(csv_parse(text))
+
```

Comment:
Add a test case for escaping a double quote, even though this is not explicitly handled by our library